src/tracing_tool_output.ml

```diff
@@ -2,21 +2,37 @@ open! Core
 open! Async
 
 module Serve = struct
-  type t =
+  type enabled =
     { port : int
-    ; perfetto_ui_base_directory : string option
+    ; perfetto_ui_base_directory : string
     }
 
+  type t =
+    | Disabled
+    | Enabled of enabled
+
   let param =
     let%map_open.Command port =
-      flag "http-port" (optional_with_default 8080 int) ~doc:"PORT http server port"
-    and perfetto_ui_base_directory =
+      let default = 8080 in
       flag
-        "perfetto-ui-base-directory"
+        "serve-port"
```

Comment:
We should consider hiding `-serve` behind whether `$PERFETTO_DIR` is set, or something. Also, it should probably take `optional_with_default 8080 int` itself, so we have one less flag.